### PR TITLE
fix(1950): refactor fetchLogs api call

### PR DESCRIPTION
**Pull request by adong**
## Context

Minor UI optimization

## Objective

- increase `LOG_RELOAD_TIMER` from `1` to `3`s
- refactored `fetchLog` to `shuttle` api service
- add tests for `shuttle` service

- upgrade to `node:12` for https://github.com/screwdriver-cd/screwdriver/issues/1722

## References

Issue: https://github.com/screwdriver-cd/screwdriver/issues/1950

Ember Data / Ember Ajax / Ember Fetch proposal
https://gist.github.com/tomdale/203ea125acd0be8b5f1f

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app/build-logs/service.js [hunk 1]
```diff
@@ -42,25 +38,24 @@ export default Service.extend({
...
+            const { response, jqXHR } = result;
+
+            if (Array.isArray(response)) {
+              lines = response;
             }
             done = started && jqXHR.getResponseHeader('x-more-data') === 'false';
           })
           // always resolve something
-          .always(() => {
+          .finally(() => {
```
Comment (adong) [inline app/build-logs/service.js]: Refactor and modernize 1 API at a time 🤣

**Comment by jithine** [comment]
looks like all unit tests are failing

**Comment by adong** [comment]
> looks like all unit tests are failing

Yep. Some old tests broke, will be updating these tests momentarily

File: app/build-logs/service.js [hunk 2]
```diff
@@ -42,25 +38,25 @@ export default Service.extend({
...
+          })
+          .then(result => {
+            const { response, jqXHR } = result;
+
+            if (Array.isArray(response)) {
+              lines = response;
             }
             done = started && jqXHR.getResponseHeader('x-more-data') === 'false';
           })
+          .catch(() => [])
```
Comment (adong) [inline app/build-logs/service.js]: I added `catch` for this interesting findings: https://stackoverflow.com/questions/47163553/try-catch-in-async-await-not-catching-error

**Pull request merged**